cmd/kubeadm/app/util/copy.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// CopyFile copies a single file
+func CopyFile(src, dst string) error {
```

Comment:
umm, I would prefer to not have to maintain this code inside of kubeadm, but I do see why it's needed right now...
what do you think about just exec'ing a `cp -r` command instead?

Reply:
Sure I have not strong opinion here, ideally I would prefer to be able to use etcd client function, but it seems the client provides snapshop, but does not provide what to do with it :)